### ISSUE TITLE
CA1862 false positive with the same variable on both sides of the comparison

### ISSUE THREAD

**Issue by xiety** [dotnet/roslyn-analyzers#7074]
### Analyzer

**Diagnostic ID**: [CA1862](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1862): `Use the 'StringComparison' method overloads to perform case-insensitive string comparisons`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 8 SDK](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

CA1862 should not be shown when the same variable is on both sides of the comparison.

### Steps To Reproduce

``` C#
if (name.ToLower() == name)
    Console.WriteLine("all letters are lowercase");
```

And the suggested fix changes the behavior completely:

``` C#
// always true now
if (name.Equals(name, StringComparison.CurrentCultureIgnoreCase))
    Console.WriteLine("all letters are lowercase");
```
